handbook/upgrade/upgrade.md

```diff
@@ -0,0 +1,280 @@
+
+    - [Setting up a SystemLink Server](https://www.ni.com/documentation/en/systemlink/latest/setup/)
+
+    - [Configuring NI Web Server](https://www.ni.com/documentation/en/ni-web-server/latest/manual/configuring-ni-web-server/)
+
+        - [Sign on with LDAP](/ldap/ldap)
+
+        - [Single Sign-on with OpenID Connect](/openid-connect/openid-connect/)
+
+    - Data Store Configuration
+
+        - [MongoDB](/data-stores/mongodb/)
+
+        - [Connecting to a Remote Mongo Database](https://www.ni.com/documentation/en/systemlink/latest/setup/remote-mongo-database/)
+
+        - [Connecting to a Remote File Share](https://www.ni.com/documentation/en/systemlink/latest/setup/remote-file-share/)
+
+        - [Uploading Files to Amazon Simple Storage Service (S3)](https://www.ni.com/documentation/en/systemlink/latest/data/uploading-files-to-amazon-s3/)
+
+        - [Connecting to a Remote PostgreSQL Database](https://www.ni.com/documentation/en/systemlink/latest/setup/remote-postgres-databse/)
+
+- Recommended
+
+    - All servers, file stores, and databases running in virtual machines
+
+    - Removable storage volume or device
+
+## Preparing for Upgrade or Migration
+
+Prior to attempting any upgrade or migration you should backup your server and any remote data stores. Failure to create backups increases the risk of unplanned downtime if issues occur during the upgrade or migration process.
+
+!!!note "Single Node versus Multi Node Configurations"
+    **Single node** refers to the SystemLink Server configuration where the application server, file storage, and databases used by SystemLink are all installed on the same Windows server. This is the default configuration for SystemLink Server.
+
+    **Multi node** refers to configurations where the file store or databases used by SystemLink are running on distinct hardware from the SystemLink application server. This is the recommended configuration for all production deployments.
+
+### NI-SystemLink-Migration Command Line Utility
+
+The workflows for upgrades and migrations make use of the SystemLink command line migration tool, `nislmigrate`. Refer to the readme at the [NI-SystemLink-Migration](https://github.com/ni/NI-SystemLink-Migration) GitHub repo for details on installing and using this tool.
+
+!!!note "Argument Flags for `nislmigrate`"
+    `nislmigrate` supports capturing data from individual services as well as all installed services using the `--all` argument flag. For brevity `--all` is used most workflows in this chapter. Depending on your needs you may replace the `--all` argument flag with one or more of the individual service argument flags.
+
+## Recommended Upgrade and Migrations Workflows for your deployment
+
+This chapter recommends workflows for the following upgrade and migration scenarios. You should plan for some downtime of your SystemLink Server. By following these recommendations this downtime can be minimized.
```

Comment:
While you should plan for some downtime of your SystemLink Server, you can minimize that downtime by following these recommendations.